pkg/apis/mysql/v1alpha1/mysqlcluster_types.go

```diff
@@ -174,6 +174,7 @@ type PodSpec struct {
 	Annotations        map[string]string         `json:"annotations,omitempty"`
 	Resources          core.ResourceRequirements `json:"resources,omitempty"`
 	Affinity           *core.Affinity            `json:"affinity,omitempty"`
+	Lifecycle          *core.Lifecycle           `json:"lifecycle,omitempty"`
```

Comment:
This is specific to `mysql` container, I sugest prefixing it with container name.
```suggestion
	MysqlLifecycle          *core.Lifecycle           `json:"mysqlLifecycle,omitempty"`
```

Reply:
Good point! Force-pushed.